scipy/signal/bsplines.py

```diff
@@ -5,12 +5,11 @@
 from scipy._lib.deprecation import _sub_module_deprecation
 
 __all__ = [  # noqa: F822
-    'spline_filter', 'bspline', 'gauss_spline', 'cubic', 'quadratic',
+    'spline_filter', 'gauss_spline',
     'cspline1d', 'qspline1d', 'cspline1d_eval', 'qspline1d_eval',
-    'logical_and', 'zeros_like', 'piecewise', 'array', 'arctan2',
-    'tan', 'arange', 'floor', 'exp', 'greater', 'less', 'add',
-    'less_equal', 'greater_equal', 'cspline2d', 'sepfir2d', 'comb',
-    'float_factorial'
+    'zeros_like', 'array', 'arctan2',
+    'tan', 'arange', 'floor', 'exp', 'greater', 'add',
+    'cspline2d', 'sepfir2d'
```

Comment:
This is removing a bunch of functions from the namespace that were not deprecated (AFAIK), or at least not related to this PR:
```
>>> before = {'logical_and', 'zeros_like', 'piecewise', 'array', 'arctan2', 'tan', 'arange', 'floor', 'exp', 'greater', 'less', 'add', 'less_equal', 'greater_equal', 'cspline2d', 'sepfir2d', 'comb', 'float_factorial'}'greater_equal', 'cspline2d', 'sepfir2d', 'comb', 'float_factorial'}
>>> after = {'zeros_like', 'array', 'arctan2', 'tan', 'arange', 'floor', 'exp', 'greater', 'add', 'cspline2d', 'sepfir2d'}
>>> before - after
{'greater_equal', 'less', 'piecewise', 'float_factorial', 'less_equal', 'comb', 'logical_and'}
>>> after - before
set()
```

Reply:
These are currently deprecated:
```
In [1]: from scipy.signal.bsplines import array
<ipython-input-1-9b99b1a2ff19>:1: DeprecationWarning: `scipy.signal.bsplines.array` is deprecated along with the `scipy.signal.bsplines` namespace. `scipy.signal.bsplines.array` will be removed in SciPy 1.13.0, and the `scipy.signal.bsplines` namespace will be removed in SciPy 2.0.0.
  from scipy.signal.bsplines import array
```